src/context/AppProvider/AppContext.tsx

```diff
@@ -135,82 +135,56 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
           const adapter = chainAdapterManager.get(chainId)
           if (!adapter) continue
 
-          switch (chainId) {
-            case ethChainId: {
-              if (!supportsETH(wallet)) continue
-              const pubkey = await adapter.getAddress({ wallet })
-              if (!pubkey) continue
-              acc.push({ [chainId]: pubkey.toLowerCase() })
-              break
-            }
-            case avalancheChainId: {
-              if (!supportsEthSwitchChain(wallet)) continue
-              const pubkey = await adapter.getAddress({ wallet })
-              if (!pubkey) continue
-              acc.push({ [chainId]: pubkey.toLowerCase() })
-              break
-            }
-            case btcChainId: {
+          const { chainNamespace, chainReference } = fromChainId(chainId)
+
+          switch (chainNamespace) {
+            case CHAIN_NAMESPACE.Bitcoin: {
               if (!supportsBTC(wallet)) continue
-              const supportedAccountTypes = (
-                adapter as unknown as bitcoin.ChainAdapter
-              ).getSupportedAccountTypes()
-              for (const accountType of supportedAccountTypes) {
-                const accountParams = utxoAccountParams(chainId, accountType, 0)
-                const { bip44Params, scriptType } = accountParams
+
+              const utxoAdapter = adapter as unknown as UtxoBaseAdapter<UtxoChainId>
+
+              for (const accountType of utxoAdapter.getSupportedAccountTypes()) {
+                const { bip44Params, scriptType } = utxoAccountParams(chainId, accountType, 0)
                 const pubkeys = await wallet.getPublicKeys([
                   {
-                    coin: 'bitcoin',
+                    coin: utxoAdapter.getCoinName(),
```

Comment:
Ah, I see now in the matching lib PR https://github.com/shapeshift/lib/pull/914/commits/ae03b705cf2d9e2d33887bc68b79ade9aeff834c

Note that we will have to lowercase the coinName property in lib, see:

https://github.com/shapeshift/hdwallet/blob/8186e4757fee8872be3bcf43f73a6c3d66c795ba/packages/hdwallet-native/src/networks.ts#L48

We had a few bugs previously where we were trying to call `wallet.getPublicKeys()` without a lowercased `coin` property

Reply:
Hmm... So it is lowercasing for native at least here: https://github.com/shapeshift/hdwallet/blob/8186e4757fee8872be3bcf43f73a6c3d66c795ba/packages/hdwallet-native/src/networks.ts#L179

It looks like keepkey has some diff logic using the protobuf messages and not sure what the firmware might be doing there. The call to `getPublicKeys` in the ChainAdapter looks like it has just been using the `coinName` as well and haven't heard of any problems? https://github.com/shapeshift/lib/blob/7006fb3946f76c306a3dad417aed71ce4acee7f1/packages/chain-adapters/src/utxo/UtxoBaseAdapter.ts#L490

Reply:
@shapeshift/operations plz ensure testing all sends with KK for monday release